### PR TITLE
TV Paint: Validate mark in and out.

### PR DESCRIPTION
**Pull request by tokejepsen**
This validator checks whether the mark in and out are enabled, and set to the database settings.

If mark in/out are disabled, TVP will return the exposure frame range. In some cases the exposure frame range and the shots frame range can differ, so it more consistent to make sure the mark in/out are set and enabled.
When rendering the last frame gets extended to the fill the mark out range.

This has a repair action for enabling and setting the mark in/out to the database asset settings.

||Pype 3 PRs|
|---|---|
|pype|https://github.com/pypeclub/pype/pull/1334|

### CODE REVIEW

File: pype/plugins/tvpaint/publish/validate_marks.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+class ValidateMarksRepair(pyblish.api.Action):
+    """Repair the marks."""
+
+    label = "Repair"
+    icon = "wrench"
+    on = "failed"
+
+    def process(self, context, plugin):
+        expected_data = ValidateMarks().get_expected_data(context)
```
Comment (iLLiCiTiT) [inline pype/plugins/tvpaint/publish/validate_marks.py:15]: Rather not create new object of `ValidateMarks` but convert `get_expected_data` to `classmethod` or `staticmethod`. Initialization may change over time and it's not save to create an object just to call a function which does not need a created object.

Comment (tokejepsen) [reply pype/plugins/tvpaint/publish/validate_marks.py]: Good point. Done

**Comment by iLLiCiTiT** [comment]
Not sure if MarkIn and MarkOut should match frameStart and frameEnd on asset (like 1001 - 1025).

**Review by iLLiCiTiT — APPROVED** [review]
Logic tested nad works.

**Review by mkolar — APPROVED** [review]

**Comment by mkolar** [comment]
@iLLiCiTiT let's port to 3.0 please

**Comment by iLLiCiTiT** [comment]
@tokejepsen `ValidateProjectSettings` should not validate frame start and frame end with this validator. Two validations of same data but one plugin has repair action.

**Comment by tokejepsen** [comment]
> @tokejepsen ValidateProjectSettings should not validate frame start and frame end with this validator. Two validations of same data but one plugin has repair action.

Good point. Done.

> this plugin can use sceneFrameStart vs. frameStart and sceneFrameEnd vs frameEnd from context instead of collecting data from workfile again

I still need the state of the mark in/out to validate, which is not collected in the workfile data. Should we collect that as well?

**Comment by iLLiCiTiT** [comment]
> Should we collect that as well?

Yes. And maybe change `sceneFrameStart` and `sceneFrameEnd` to `sceneMarkIn` and `sceneMarkOut` to make it clear? I did store the values as `frameStart` and `frameEnd` because values were used for whole publishing (extract review etc.) which does not make sence if they have `scene` prefix.

**Comment by tokejepsen** [comment]
> Yes. And maybe change sceneFrameStart and sceneFrameEnd to sceneMarkIn and sceneMarkOut to make it clear? I did store the values as frameStart and frameEnd because values were used for whole publishing (extract review etc.) which does not make sence if they have scene prefix.

Cool, done.

**Review by iLLiCiTiT — APPROVED** [review]
Approved

**Pull request merged**